### ISSUE TITLE
demandlib error in creation of h array used in the BDEW heatprofile method

### ISSUE THREAD

**Issue by birgits** [oemof/oemof#129]
The following line should be changed from

`h = np.array(SF_mat)[np.array(range(0, 8760))[:], (get_temperature_interval(df) - 1)[:]]`

to

`h = np.array(SF_mat)[np.array(range(0, 8760))[:], (get_temperature_interval(df) + 2)[:]]`

since the temp_intervall columns in the SF_mat array don't start in column 0 but in column 3.
I created a bugfix branch bugfix/bdew_heatprofile where I implemented that change.
@simnh: What do you think?

**Comment by Simn** [comment]
You got the wrong guy, but hi anyway!

**Comment by simnh** [comment]
I am not really into that functionality, so I you think it's correct feel free to merge (if functionalities still work)

**Comment by birgits** [comment]
Testing the bugfix with a commercial type produced another error so I now fixed it by dropping unnecessary columns from the SF_mat dataframe. Also, since there is no parameter building_class for commercial types I changed it to be a keyword argument with the default value 0, since building_class for commercial types was set to 0 in the shlp_hour_factors.csv.
The changes passed the validation with the heat_demand_example.py for all residential and commercial types so I will push it to the dev.
